tests/conftest.py

```diff
@@ -235,8 +242,11 @@ def project_no_init(tmp_path, mocker):
     do_use(p, getattr(sys, "_base_executable", sys.executable))
     with temp_environ():
         os.environ.pop("VIRTUAL_ENV", None)
-        os.environ.pop("PYTHONPATH", None)
         os.environ.pop("PEP582_PACKAGES", None)
+        pythonpath = os.environ.pop("PYTHONPATH", "")
+        pythonpath = remove_pep582_path_from_pythonpath(pythonpath)
```

Comment:
We could make this simpler for testing, just clean the `PYTHONPATH` here. No need to keep the old value. Test case can provide its value when running

Reply:
We could not assume the development environment of others not taking PYTHONPATH in usage. Some may use nix to do development. And nix is using PYTHONPATH to import packages that may essential to pdm testings. For example, the pip package of python3.9 for downloading wheel files for testing.

Reply:
I didn't know the approach used by nix.